### DISCUSSION TITLE
Scenario Timeout

### DISCUSSION THREAD

**Discussion by lauji2k19** [VIP-Logic/NBomber#611] [Q&A]
Hi,

I am using NBomber V4.1.2 and facing an issue where the scenario is stopped before all steps are able to complete. I figure this is because the load test has several sequential requests and exceeds the scenario duration. We first make a sequence of 4 HTTP requests which take anywhere between 1-10 seconds each to respond. Then, the fifth step is asynchronous so we ping an endpoint every second for 2 minutes maximum. Here's a rough time estimate for each step:

1. ~1 sec. 45/45 requests executed.
2. ~1-4 secs. 45/45 requests executed (1 failed, which is OK so I expect the total to now be 44).
3. ~4-10 secs. 26/44 requests executed.
4. ~4-6 secs. 21/44 requests executed.
5. ~15-30 secs. 0/44 requests executed.

The test is run with the load simulation:
`Simulation.Inject(rate: 3, interval: TimeSpan.FromSeconds(1), during: TimeSpan.FromSeconds(15)`

The goal of the test is to send 3 RPS for 15 seconds and see how long it takes for the system to complete the whole process and identify any bottlenecks, but I can't do this since the scenario ends before all steps execute. Is there a way to configure the test to only stop either if all iterations complete, or at a max cutoff, say, 10 minutes?

**Comment by AntyaDev** [comment]
Hi @lauji2k19 
Answered here: https://github.com/PragmaticFlow/NBomber/issues/613#issuecomment-1679121662
